docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,333 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _dissociated-ipc:
+
+========================
+Dissociated IPC Protocol
+========================
+
+.. warning::
+
+    Experimental: The Dissociated IPC Protocol is experimental in its current
+    form. Based on feedback and usage the protocol definition may change until
+    it is fully standardized.
+
+Rationale
+=========
+
+The :ref:`Arrow IPC format <format-ipc>` describes a protocol for transferring
+Arrow data as a stream of record batches. This protocol expects a continuous
+stream of bytes divided into discrete messages (using a length prefix and
+continuation indicator). Each discrete message consists of two portions:
+
+* A `Flatbuffers`_ header message
+* A series of bytes consisting of the flattened and packed body buffers (some
+message types, like Schema messages, do not have this section)
+  - This is referred to as the *message body* in the IPC format spec.
+
+For most cases, the existing IPC format as it currently exists is extremely efficient:
+
+* Receiving data in the IPC format allows zero-copy utilization of the body
+buffer bytes, no deserialization is required to form Arrow Arrays
+* An IPC (Feather) file can be memory-mapped because it is location agnostic
+and the bytes of the file are exactly what is expected in memory.
+
+However, there are use cases that aren't handled by this:
+
+* Constructing the IPC record batch message requires allocating a contiguous
+chunk of bytes and copying all of the data buffers into it, packed together
+back-to-back. You currently cannot zero-copy **create** IPC messages.
+* If the Arrow data is located in a shared-memory location, there is no standard
+way to share the handle to the shared-memory across processes or transports that
+allow for remote memory accessing.
+* Arrow data located on a non-CPU device (such as a GPU) cannot be sent using
+Arrow IPC without having to copy the data back to the host device or copying
+the flatbuffer metadata bytes into device memory.
+  - By the same token, receiving IPC messages into device memory would require
+  performing a copy of the flatbuffer metadata back to the host CPU device.
```

Comment:
Might be worth explicitly stating that this is because IPC colocates/interleaves data and metadata?